static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -0,0 +1,1033 @@
+  matchingArtifactName: string;
+  projectHasUploadedArtifacts: boolean;
+  releaseName: string | null;
+  releaseSourceMapReference: string | null;
+  sdkDebugIdSupport: 'full' | 'needs-upgrade' | 'unofficial-sdk';
+  sourceFileReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceFileScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  sourceMapReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceMapScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  stackFrameDebugId: string | null;
+  stackFramePath: string | null;
+  uploadedSomeArtifact: boolean;
+  uploadedSomeArtifactToRelease: boolean;
+  uploadedSomeArtifactWithDebugId: boolean;
+  uploadedSourceFileWithCorrectDebugId: boolean;
+  uploadedSourceMapWithCorrectDebugId: boolean;
+  sdkVersion?: string;
+}
+
+interface SourceMapsDebuggerModalProps extends ModalRenderProps {
+  sourceResolutionResults: SourceResoultionResults;
+}
+
+export function SourceMapsDebuggerModal({
+  Body,
+  Header,
+  Footer,
+  sourceResolutionResults,
+}: SourceMapsDebuggerModalProps) {
+  const theme = useTheme();
+
+  let debugIdProgress = 0;
+  if (sourceResolutionResults.sdkDebugIdSupport === 'full') {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.stackFrameDebugId !== null) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceFileWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceMapWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  const debugIdProgressPercent = debugIdProgress / 4;
+
+  let releaseNameProgress = 0;
+  if (sourceResolutionResults.releaseName !== null) {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.uploadedSomeArtifactToRelease) {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.sourceFileReleaseNameFetchingResult === 'found') {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.sourceMapReleaseNameFetchingResult === 'found') {
+    releaseNameProgress++;
+  }
+  const releaseNameProgressPercent = releaseNameProgress / 4;
+
+  let scrapingProgress = 0;
+  if (sourceResolutionResults.sourceFileScrapingStatus.status === 'found') {
+    scrapingProgress++;
+  }
+  if (sourceResolutionResults.sourceMapScrapingStatus.status === 'found') {
+    scrapingProgress += 4;
+  }
+  const scrapingProgressPercent = scrapingProgress / 5;
+
+  const [activeTab, setActiveTab] = useState<'debug-ids' | 'release' | 'fetching'>(() => {
+    const possibleTabs = [
+      {tab: 'debug-ids', progress: debugIdProgressPercent},
+      {tab: 'release', progress: releaseNameProgressPercent},
+      {tab: 'fetching', progress: scrapingProgressPercent},
+    ] as const;
+
+    return possibleTabs.reduce((prev, curr) => {
+      if (curr.progress > prev.progress) {
+        return curr;
+      }
+      return prev;
```

Comment:
mega nit: Can we ternery?

```suggestion
      return curr.progress > prev.progress ? curr : prev;
```